subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -1759,24 +1759,35 @@ int nrf_cloud_format_cell_pos_req_json(struct lte_lc_cells_info const *const inf
 
 	lte_array = cJSON_AddArrayToObjectCS(req_obj_out, NRF_CLOUD_CELL_POS_JSON_KEY_LTE);
 	if (!lte_array) {
+		err = -ENOMEM;
 		goto cleanup;
 	}
 
-	/* Add the current cell to the array */
-	lte_obj = add_lte_inf(lte_array, &inf->current_cell);
-	if (!lte_obj) {
-		goto cleanup;
-	}
+	/* Add the current cell to the array; if using a GCI search type, sometimes
+	 * there is no current cell.
+	 */
+	if (inf->current_cell.id != LTE_LC_CELL_EUTRAN_ID_INVALID) {
```

Comment:
What will happen if there are no cells whatsoever -- Should we raise an error in that case, or in some other way abort sending the request?

Reply:
Look further below, line 1786.

"No current cell" in this context means that there is no specific serving cell, I guess because we are in RRC IDLE mode.  But once it goes back to RRC CONN mode, there will be a serving cell.

Reply:
Ah, got it, I see now, thank you! 

This makes me realize that the `Skip GCI cells if not present`, comment should probably be updated:

```
/* If GCI cells are not present, skip appending them, and verify that we at least have a serving cell */
if (!inf->gci_cells_count || !inf->gci_cells) {
		return 0;
		if (inf->current_cell.id == LTE_LC_CELL_EUTRAN_ID_INVALID) {
			err = -ENODATA;
		} else {
		      
			err = 0;
		}
		goto cleanup;
	}
```

Reply:
Another approach would be to introduce a `total_cells`, and raise `-ENODATA` at the end of the function if that tally is zero -- purely for the sake of making it a little clearer to a prospective reader. Up to you whether you prefer to do that or just update the comment

Reply:
The serving cell is retained also in RRC idle, that's the cell modem is listening to pagings from.

If there's no serving cell, then modem must have lost the cell it was camped on and it hasn't yet found a new one (or the same one again). While it may be possible to resolve the location also without knowing the serving cell, it's not possible to send the request to the network without one. But probably it's better not to limit the usability of the scan results and let modem try to find a cell and send the request to the network.

Reply:
OK.  What do you suggest I do instead?

Here is what happens when I do not have a serving cell, but I do have GCI cells:
```
[00:16:28.649,047] <inf> nrf_cloud_rest_cell_location_sample: Requesting neighbor cell measurement
[00:16:28.697,448] <inf> nrf_cloud_rest_cell_location_sample: Waiting for measurement results...
%NCELLMEAS: 0,"0509260F","310410","900A",65535,0,5110,309,69,14,1060752,0,0,"0D8A6017","310260","352B",65535,0,5035,203,72,16,1060775,0,0,"0D8A6015","310260","352B",65535,0,5035,495,70,13,1060775,0,0,"00147004","311480","0505",65535,0,5230,75,70,26,1060798,0,0
[00:17:41.303,894] <dbg> lte_lc: at_handler_ncellmeas_gci: %NCELLMEAS GCI notification parsing starts
[00:17:41.315,551] <dbg> lte_lc: at_handler_ncellmeas_gci: parse_ncellmeas_gci returned 0
[00:17:41.324,096] <dbg> lte_lc: at_handler_ncellmeas_gci: Neighbor cell count: 0, GCI cells count: 4
[00:17:41.333,740] <dbg> lte_lc_helpers: event_handler_list_dispatch: Dispatching events:
[00:17:41.342,346] <dbg> lte_lc_helpers: event_handler_list_dispatch:  - handler=0x0001FDA5
[00:17:41.351,135] <wrn> nrf_cloud_rest_cell_location_sample: No neighbor cells were measured
[00:17:41.360,382] <inf> nrf_cloud_rest_cell_location_sample: Received measurements for 4 GCI cells
[00:17:41.369,842] <dbg> lte_lc_helpers: event_handler_list_dispatch: Done
[00:17:41.377,258] <inf> nrf_cloud_rest_cell_location_sample: Current cell info: Cell ID: 4294967295, TAC: 0, MCC: 0, MNC: 0
[00:17:41.388,885] <inf> nrf_cloud_rest_cell_location_sample: Performing multi-cell request with 0 neighbor cells and 4 GCI cells
+CSCON: 1
[00:17:42.069,274] <dbg> lte_lc: at_handler_cscon: +CSCON notification
[00:17:42.076,324] <dbg> lte_lc_helpers: event_handler_list_dispatch: Dispatching events:
[00:17:42.084,899] <dbg> lte_lc_helpers: event_handler_list_dispatch:  - handler=0x0001FDA5
[00:17:42.093,658] <inf> nrf_cloud_rest_cell_location_sample: RRC mode: connected
[00:17:42.101,562] <dbg> lte_lc_helpers: event_handler_list_dispatch: Done
[00:17:47.728,790] <inf> nrf_cloud_rest_cell_location_sample: Cellular location request fulfilled with single-cell
[00:17:47.739,532] <inf> nrf_cloud_rest_cell_location_sample: Lat: 45.523117, Lon: -122.685517, Uncertainty: 300

Reply:
I think the implementation is fine, just commented that if there's no serving cell, it might not be possible to send the request to the network, that's all.

Reply:
Ah, yes, of course.  So far in testing it always reconnects and sends the request, though it is hard to reproduce this no-serving-cell issue.  It happened a number of times that one day, but not since.